### ISSUE TITLE
How to use custom WASM runtime?

### ISSUE THREAD

**Issue by yjhmelody** [as-pect/as-pect#363]
It seems to be using node's wasm runtime. But I need to run other wasm runtime for tests

**Comment by willemneal** [comment]
Yes currently this project externalizes a lot of the runtime logic in JS.  There have been plans in the past to make test binaries compatible with Wasi/minimize the required imports needed to executed and report tests.

**Comment by jtenner** [comment]
I think that using WASI is probably moot, and a new specialized tool that generates tests with wasi should be created.

@willemneal I believe that we should refactor the strict equals transform, comparison functions, and `expect()` into their own library for this purpose.
